cupy/lib/scimath.py

```diff
@@ -0,0 +1,541 @@
+    >>> cc.dtype
+    dtype('complex64')
+    >>> c *= 2; c
+    array([2.+0.j,  4.+0.j,  6.+0.j], dtype=complex64)
+    >>> cc
+    array([1.+0.j,  2.+0.j,  3.+0.j], dtype=complex64)
+
+    See Also
+    --------
+    numpy.lib.scimath._tocomplex
+
+    """
+    if issubclass(arr.dtype.type,
+                  (cupy.single, cupy.byte,
+                   cupy.short, cupy.ubyte,
+                   cupy.ushort, cupy.csingle)):
+        return arr.astype(cupy.csingle)
+    else:
+        return arr.astype(cupy.cdouble)
+
+
+def _fix_real_lt_zero(x):
+    """Convert `x` to complex if it has real, negative components.
+    Otherwise, output is just the array version of the input
+    (via cupy.asarray).
+
+    Parameters
+    ----------
+    x : array_like
+
+    Returns
+    -------
+    array : cupy.ndarray
+
+    Examples
+    --------
+    >>> cupy.lib.scimath._fix_real_lt_zero([1,2])
+    array([1, 2])
+    >>> cupy.lib.scimath._fix_real_lt_zero([-1,2])
+    array([-1.+0.j,  2.+0.j])
+
+    See Also
+    --------
+    numpy.lib.scimath._fix_real_lt_zero
+
+    """
+    x = cupy.asarray(x)
+    if cupy.any(cupy.isreal(x) & (x < 0)):
+        x = _tocomplex(x)
+    return x
+
+
+def _fix_int_lt_zero(x):
+    """Convert `x` to double if it has real, negative components.
+    Otherwise, output is just the array version of the input
+    (via cupy.asarray).
+
+    Parameters
+    ----------
+    x : array_like
+
+    Returns
+    -------
+    array : cupy.ndarray
+
+    Examples
+    --------
+    >>> cupy.lib.scimath._fix_int_lt_zero([1,2])
+    array([1, 2])
+    >>> cupy.lib.scimath._fix_int_lt_zero([-1,2])
+    array([-1.,  2.])
+
+    See Also
+    --------
+    numpy.lib.scimath._fix_int_lt_zero
+
+    """
+    x = cupy.asarray(x)
+    if cupy.any(cupy.isreal(x) & (x < 0)):
+        x = x * 1.0
+    return x
+
+
+def _fix_real_abs_gt_1(x):
+    """Convert `x` to complex if it has real components x_i with abs(x_i)>1.
+    Otherwise, output is just the array version of the input
+    (via cupy.asarray).
+
+    Parameters
+    ----------
+    x : array_like
+
+    Returns
+    -------
+    array : cupy.ndarray
+
+    Examples
+    --------
+    >>> cupy.lib.scimath._fix_real_abs_gt_1([0,1])
+    array([0, 1])
+    >>> cupy.lib.scimath._fix_real_abs_gt_1([0,2])
+    array([0.+0.j, 2.+0.j])
+
+    See Also
+    --------
+    numpy.lib.scimath._fix_real_abs_gt_1
+
+    """
+    x = cupy.asarray(x)
+    if cupy.any(cupy.isreal(x) & (abs(x) > 1)):
+        x = _tocomplex(x)
+    return x
+
+
+def sqrt(x):
+    """
+    Compute the square root of x.
+    For negative input elements, a complex value is returned
+    (unlike `cupy.sqrt` which returns NaN).
+
+    Parameters
+    ----------
+    x : array_like
+       The input value(s).
+
+    Returns
+    -------
+    out : cupy.ndarray
+       The square root of `x`.
+
+    Examples
+    --------
+    For real, non-negative inputs this works just like `cupy.sqrt`:
```

Comment:
The examples block looks broken. A blank line fixes it maybe?
https://cupy--7295.org.readthedocs.build/en/7295/reference/generated/cupy.lib.scimath.sqrt.html#cupy.lib.scimath.sqrt

Reply:
Yes, a blank line should fix it.